### PR TITLE
feat(exec_mode): add exec_mode to start_webserver

### PR DESCRIPTION
**Pull request by psavery**
This adds three exec_modes to start_webserver: "main", "task", and "coroutine". The "main" mode
is the previous behavior of blocking python execution and closing the loop afterwards. The "task"
mode creates a task and returns it (does not block execution and does not close the loop). The
"coroutine" just returns the coroutine without starting it. The "task" is particularly useful for
running inside a jupyter notebook, which already has an asyncio event loop running, and for which
we do not want to block python execution.

### CODE REVIEW

File: python/src/wslink/server.py [hunk 1]
```diff
@@ -166,7 +166,8 @@ def create_webserver(server_config, backend="aiohttp"):
 # and start it.
 # =============================================================================
 def start_webserver(
-    options, protocol=wsl.ServerProtocol, disableLogging=False, backend="aiohttp"
+    options, protocol=wsl.ServerProtocol, disableLogging=False, backend="aiohttp",
+    exec_mode="main"
```
Comment (jourdain) [inline python/src/wslink/server.py:170]: can you add a **kwargs

Comment (jourdain) [reply python/src/wslink/server.py:170]: that way if we pass things we don't care we are still ok

Comment (psavery) [reply python/src/wslink/server.py]: Done.

**Review by jourdain — APPROVED** [review]
LGTM

**Pull request merged**